City-of-Helsinki/berth-reservations
# VEN-812 | Handle repeated payment attempts

## Description :sparkles:
* Add an `OrderToken` model to store the payment tokens retrieved from Bambora
* Check if a valid token exists before trying to initiate the payment flow

## Issues :bug:
### Closes :no_good_woman:
**[VEN-812](https://helsinkisolutionoffice.atlassian.net/browse/VEN-812):** Handle repeated payment attempts

## Testing :alembic:
### Automated tests :gear:️
```shell
$ python manage.py payments/tests/test_bambora_payform.py::test_initiate_duplicated_payment
$ python manage.py payments/tests/test_bambora_payform.py::test_initiate_duplicated_payment_new_token_after_expiry
```

### Manual testing :construction_worker_man:
1. Create an order
2. Go to `https://venepaikka.test.kuva.hel.ninja/fi/payment?order_number=XXXXXX` for your order
3. Click `Jatka maksamaan`
4. Close the Bambora tab
5. Open the Customer UI page again and click `Jatka maksamaan` again
6. The Bambora order should still be payable
